sample-app/src/hooks/useEntityPreviews.tsx

```diff
@@ -25,12 +25,24 @@ export function useEntityPreviews(headlessId: string, debounceTime: number):[ En
       headlessId
     });
   }
+  const isMounted = useRef<boolean>(false);
+  useEffect(() => {
```

Comment:
same questions here (I assume the two are linked but maybe not...?)

Reply:
similar to my response above, `useDebouncedFunction` invoke an async function that would wait for a response, before calling state update functions. There would be instances where during the wait for the response, the component unmounted, then it would proceed to execute the state update functions `setVerticalResultsArray` `setLoadingState` which would cause an error.

Reply:
are universal searches firing when selecting a vertical link? let's move this isMounted handling into useDebouncedSearch, since it seems logical that a debounced function might result in async behavior that we would need this handling for.

Reply:
it's more so an edge case/race condition thing. A vertical link is under an option that would display entity preview. So when user navigate with key, this could happen:
1) arrow down => focus on an option (trigger universal search for entity preview)
2)arrow down + enter => select a vertical link and navigate (async done waiting and start executing the other state update functions)
I can't move it into useDebouncedSearch because the issue is within the function that's pass into the useDebouncedSearch. Since the state update occurs in useEntityPreview, I think it makes since to have it in here.  useDebouncedSearch is not aware of those state updates and is only in charge of debounce stuff

Reply:
Did a little research and this warning has actually been removed from the react codebase, so it should probably be gone in react 18?
https://medium.com/doctolib/react-stop-checking-if-your-component-is-mounted-3bb2568a4934
[the linked pr](https://github.com/facebook/react/pull/22114)
It makes me feel a little better that a lot of other people run into this issue, and that our case here is considered a false positive (I think) in terms of it being a memory leak. 
I'm fine with either extracting isMounted into it's own hook, or just letting this errors go unchecked if we're sure they're not memory leaks (which I'm pretty sure they're not...?)

Reply:
It will probably be at least a few months until react 18 so leaning towards just hiding the errors with a hook, and then when react 18 finally comes it should be easy to find usages of the hook